internal/git/pr.go

```
@@ -0,0 +1,41 @@
+package git
+
+import (
+	"fmt"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// GetPRInfo phrases GitHub PR URL: returns owner, repo, prNumber, error
+func GetPRInfo(prURL string) (string, string, int, error) {
+	if !IsValidGitHubPRURL(prURL) {
+		return "", "", 0, fmt.Errorf("the PR url is invalid: %s", prURL)
+	}
+
+	u, err := url.Parse(prURL)
+	if err != nil {
+		panic(err)
+	}
+
+	parts := strings.Split(strings.Trim(u.Path, "/"), "/")
+	if len(parts) < 4 || parts[2] != "pull" {
+		return "", "", 0, fmt.Errorf("not a valid PR URL %s", prURL)
```

The function panics on URL parsing error instead of returning it properly. Replace panic(err) with a proper error return to maintain consistent error 
handling.